File: Design-Patterns-Using-CSharp/Basics OOPS/CompositionRelationship.cs
using System;
using System.Collections.Generic;

class Tyre
{
    private string type;

    public Tyre(string type)
    {
        this.type = type;
    }

    public string GetType()
    {
        return type;
    }
}

class Car
{
    private string model;
    private List<Tyre> tyres;

    public Car(string model)
    {
        this.model = model;
        this.tyres = new List<Tyre>();
        for (int i = 0; i < 4; i++)
        {
            this.tyres.Add(new Tyre("MRF"));
        }
    }

    public void Display()
    {
        Console.WriteLine("Car: " + model + ", Tyre: " + tyres[0].GetType());
    }
}

public class CompositionRelationship
{
    public static void Main(string[] args)
    {
        Car car = new Car("BMW");
        car.Display();
    }
}

File: Design-Patterns-Using-Cpp/Basics OOPS/CompositionRelationship.cpp
#include <iostream>
#include <vector>

class Tyre {
private:
    std::string type;

public:
    Tyre(const std::string& type) : type(type) {}

    std::string getType() const {
        return type;
    }
};

class Car {
private:
    std::string model;
    std::vector<Tyre> tyres;

public:
    Car(const std::string& model) : model(model) {
        for (int i = 0; i < 4; i++) {
            tyres.push_back(Tyre("MRF"));
        }
    }

    void display() const {
        std::cout << "Car: " << model << ", Tyre: " << tyres[0].getType() << std::endl;
    }
};

int main() {
    Car car("BMW");
    car.display();

    return 0;
}
/*
Car: BMW, Tyre: MRF
*/